src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx

```diff
@@ -35,6 +41,30 @@ const FaultEventMenu = ({ shapeToolData, onEventUpdated, refreshTree, rootIri }:
   const [quantity, setQuantity] = useState<number | undefined>(undefined);
   const [schematicDesignation, setSchematicDesignation] = useState<string | undefined>(undefined);
 
+  const [snsPredictedFailureRate, setSnsPredictedFailureRate] = useState<number | undefined>(undefined);
+  const [snsManuallyDefinedFailureRate, setSnsManuallyDefinedFailureRate] = useState<number | undefined>(undefined);
+  const [selectedRadioButton, setSelectedRadioButton] = useState<string>(RadioButtonType.Predicted);
+
+  const handleManuallyDefinedFailureRateChange = (event) => {
+    const inputValue = event.target.value;
+    const regex = /^[0-9]*\.?[0-9]*$/;
```

Comment:
For manual value property we need to support not only decimal number in this format but also in a scientific format that includes exponent (e.g. as 1.23E3)

Reply:
https://github.com/kbss-cvut/fta-fmea-ui/issues/368